cpu/msp430_common/malloc.c

```diff
@@ -1,486 +0,0 @@
-            irq_restore(state);
-            return ptr;
-        }
-        fp2 = (struct __freelist *)cp;
-        fp2->sz = fp1->sz - len - sizeof(size_t);
-        fp1->sz = len;
-        free(&(fp2->nx));
-        irq_restore(state);
-        return ptr;
-    }
-
-    /*
-     * If we get here, we are growing.  First, see whether there
-     * is space in the free list on top of our current chunk.
-     */
-    incr = len - fp1->sz;
-    cp = (char *)ptr + fp1->sz;
-    fp2 = (struct __freelist *)cp;
-    for (s = 0, ofp3 = 0, fp3 = __flp;
-         fp3;
-         ofp3 = fp3, fp3 = fp3->nx) {
-        if (fp3 == fp2 && fp3->sz + sizeof(size_t) >= incr) {
-            /* found something that fits */
-            if (fp3->sz + sizeof(size_t) - incr > sizeof(struct __freelist)) {
-                /* split off a new freelist entry */
-                cp = (char *)ptr + len;
-                fp2 = (struct __freelist *)cp;
-                fp2->nx = fp3->nx;
-                fp2->sz = fp3->sz - incr;
-                fp1->sz = len;
-            }
-            else {
-                /* it just fits, so use it entirely */
-                fp1->sz += fp3->sz + sizeof(size_t);
-                fp2 = fp3->nx;
-            }
-            if (ofp3) {
-                ofp3->nx = fp2;
-            }
-            else {
-                __flp = fp2;
-            }
-            irq_restore(state);
-            return ptr;
-        }
-        /*
-         * Find the largest chunk on the freelist while
-         * walking it.
-         */
-        if (fp3->sz > s) {
-            s = fp3->sz;
-        }
-    }
-    /*
-     * If we are the topmost chunk in memory, and there was no
-     * large enough chunk on the freelist that could be re-used
-     * (by a call to malloc() below), quickly extend the
-     * allocation area if possible, without need to copy the old
-     * data.
-     */
-    if (__brkval == (char *)ptr + fp1->sz && len > s) {
-        cp1 = __malloc_heap_end;
-        cp = (char *)ptr + len;
-        if (cp1 == NULL) {
-            char *stack_pointer;
-            asmv("mov r1, %0" : "=r"(stack_pointer));
-            cp1 = stack_pointer - __malloc_margin;
-        }
-        if (cp < cp1) {
-            __brkval = cp;
-            fp1->sz = len;
-            irq_restore(state);
-            return ptr;
-        }
-        /* If that failed, we are out of luck. */
-        irq_restore(state);
-        return 0;
-    }
-
-    /*
-     * Call malloc() for a new chunk, then copy over the data, and
-     * release the old region.
-     */
-    if ((memp = malloc(len)) == NULL) {
-        irq_restore(state);
-        return 0;
-    }
-    memcpy(memp, ptr, fp1->sz);
-    free(ptr);
-    irq_restore(state);
-    return memp;
-}
-
-void *
-calloc(size_t nele, size_t size)
-{
-    void *p;
-
-    if ((p = malloc(nele * size)) == NULL) {
-        return 0;
-    }
-    memset(p, 0, nele * size);
-    return p;
-}
-
-void heap_stats(void)
```

Comment:
We need a new implementation of this function, at least the "not supported" part. That's why `tests/heap_stats` is not working.

Reply:
Yes. I'll get it back.

Reply:
Ah no, it should be able to use the default one now. I've removed the "HAVE_HEAP_STATS" define.

Reply:
Yes, if the platform is using module `newlib_syscalls_default`, the default implementation can be used.